### ISSUE TITLE
ISSUE : ContractNotifications multiple Filters & Groups

### ISSUE THREAD

**Issue by dreej-net** [panthernet/ThunderED#58]
Tests done :
- Issuing Courier contract to Corporation.
- From Citadel to Jita trade Station.
- No reward / No Collateral
- Description not null.  

Result obtained :
- Notifications of FRESH contracts are not display in both channel, only the one set for "group1".
- Notifications of other statuses (not tested all of them though) are displayed correctly in both channels.
- Newly created issued contracts are re-notifying the very last notification (even if contract_id is not the same id as the one triggering the notification), in both channels.

Errors :
 - No consol errors.
 - ESI.log @issued_contract_timestamp shows : 
```
04/22/2019 14:30:24 [   Error]: [try: 0][ContractNotif] Potential BadRequest request failure: https://esi.evetech.net/latest/universe/stations/1030055977622/?datasource=tranquility&language=en-us
```
- Other .log files show no error @timestamp.  

... with Groups set-up like so in settings.json :

```

  "ContractNotificationsModule": {
    "CheckIntervalInMinutes": 1,
    "MaxTrackingCount": 150,
    "Groups": {
      "group1": {
        "CharacterIDs": [
          74790907
        ],
        "Filters": {
          "clients": {
            "Availability": [
              "corporation"
            ],
            "Types": [
              "courier"
            ],
            "Statuses": [
              "finished_issuer",
              "finished_contractor",
              "finished",
              "cancelled",
              "rejected",
              "failed",
              "deleted",
              "reversed",
              "in_progress",
              "outstanding"
            ],
            "DiscordChannelId": 569258723709660599,
            "FeedIssuedBy": false,
            "FeedIssuedTo": true,
            "ShowIngameOpen": true,
            "ShowOnlyBasicDetails": true,
            "RedirectByIdInDescription": true,
            "PostToChannelIfRedirected": false
          }
        },
        "FeedPersonalContracts": false,
        "FeedCorporateContracts": true,
        "ButtonText": "Default Contracts Auth",
        "DefaultMention": "@here"
      },
      "group2": {
        "CharacterIDs": [
          74790907
        ],
        "Filters": {
          "logisticiens": {
            "Availability": [
              "corporation"
            ],
            "Types": [
              "courier"
            ],
            "Statuses": [
              "finished_issuer",
              "finished_contractor",
              "finished",
              "cancelled",
              "rejected",
              "failed",
              "deleted",
              "reversed",
              "in_progress",
              "outstanding"
            ],
            "DiscordChannelId": 569575843885726761,
            "FeedIssuedBy": false,
            "FeedIssuedTo": true,
            "ShowIngameOpen": true,
            "ShowOnlyBasicDetails": false,
            "RedirectByIdInDescription": false,
            "PostToChannelIfRedirected": false
          }
        },
        "FeedPersonalContracts": false,
        "FeedCorporateContracts": true,
        "ButtonText": "Default Contracts Auth",
        "DefaultMention": "@everyone"
      }
    }
  },
```
________

Same test done but with multiple filters in same group.

Result obtained :
- Fresh contracts notifs are displayed in first filter channel only.  
- Status changes notifs are displayed in both channels.
- Fresh contracts issued are not notified.

Errors :
- Error in Console : ```[Critical] [ContractNotif] : Contracts```
- ContractNotif.log shows : 

```
04/22/2019 15:22:45 [Critical]: Contracts 
System.ArgumentOutOfRangeException: Index must be within the bounds of the List.
Parameter name: index
   at System.Collections.Generic.List`1.Insert(Int32 index, T item)
   at ThunderED.Modules.ContractNotificationsModule.ProcessContracts(Boolean isCorp, ContractNotifyGroup group, Int64 characterID, String token) in E:\ThunderED-master\ThunderED\Modules\ContractNotificationsModule.cs:line 261
   at ThunderED.Modules.ContractNotificationsModule.Run(Object prm) in E:\ThunderED-master\ThunderED\Modules\ContractNotificationsModule.cs:line 163
```  
- ESI.log shows : 
```
04/22/2019 15:22:43 [   Error]: [try: 0][ContractNotif] Potential BadRequest request failure: https://esi.evetech.net/latest/universe/stations/1030055977622/?datasource=tranquility&language=en-us
04/22/2019 15:22:44 [   Error]: [try: 0][ContractNotif] Potential BadRequest request failure: https://esi.evetech.net/latest/universe/stations/1030055977622/?datasource=tranquility&language=en-us
```
- No other errors in other .log files.  

Note : 
New contract has

...with settings.json set-up like so :

```
  "ContractNotificationsModule": {
    "CheckIntervalInMinutes":  1,
    "MaxTrackingCount":  150, 
    "Groups": {
      "group1": {
        "CharacterIDs": [ 74790907 ],
        "ButtonText": "Default Contracts Auth",
        "DefaultMention": "@here",
        "FeedPersonalContracts": false,
        "FeedCorporateContracts": true,
        "Filters": {
          "clientsFilter": {
            "FeedIssuedBy": false,
            "FeedIssuedTo": true,
            "Statuses": [ "finished_issuer", "finished_contractor", "finished", "cancelled", "rejected", "failed", "deleted", "reversed", "in_progress", "outstanding" ],
            "Availability": [ "corporation" ],
            "Types": [ "unknown", "courier" ],
            "ShowIngameOpen": true,
            "DiscordChannelId": 569258723709660599,
            "ShowOnlyBasicDetails": true,
            "RedirectByIdInDescription": true,
            "PostToChannelIfRedirected": false
          },
          "logisticiensFilter": {
            "FeedIssuedBy": false,
            "FeedIssuedTo": true,
            "Statuses": [ "finished_issuer", "finished_contractor", "finished", "cancelled", "rejected", "failed", "deleted", "reversed", "in_progress", "outstanding" ],
            "Availability": [ "corporation" ],
            "Types": [ "unknown", "courier" ],
            "ShowIngameOpen": true,
            "DiscordChannelId": 569575843885726761,
            "ShowOnlyBasicDetails": false,
            "RedirectByIdInDescription": false,
            "PostToChannelIfRedirected": false
          }
        }
      }
    } 
  },
```

**Comment by dreej-net** [comment]
Well it seems, that you need to auth 2 times if you use multiple groups

**Comment by dreej-net** [comment]
Seems to be solved with 1.3.9.

**Comment by panthernet** [comment]
Okay, let me know if there's something else going weird with contracts,
